### PR TITLE
CO-3865 import bankstatement with currencies, affect other bankstatem…

### PR DESCRIPTION
**Pull request by TheRaphael0000**
…ent line

fetch the journal related to a bank account and pass it to the context

### CODE REVIEW

File: account_bank_statement_import_camt_oca/models/parser.py [hunk 1]
```diff
@@ -199,6 +200,17 @@ def parse_statement(self, ns, node):
                 './ns:Acct/ns:Id/ns:Othr/ns:Id',
             ], result, 'account_number'
         )
+
+        if result["account_number"]:
```
Comment (ecino) [inline account_bank_statement_import_camt_oca/models/parser.py:204]:
```suggestion
        if result.get("account_number"):
```

File: account_bank_statement_import_camt_oca/models/parser.py [hunk 2]
```diff
@@ -199,6 +200,17 @@ def parse_statement(self, ns, node):
...
+            # find the journal related to this account
+            journals = self.env["account.journal"].search([])
+            trans_table = str.maketrans("", "", string.whitespace)
+            account_number_ = result["account_number"].translate(trans_table)
+            journal = journals.filtered(
+                lambda x: x.bank_acc_number and x.bank_acc_number.translate(trans_table) == account_number_)
```
Comment (ecino) [inline account_bank_statement_import_camt_oca/models/parser.py:210]: bank_acc_number is a required field so it is always set in theory.
```suggestion
                lambda x: x.bank_acc_number.translate(trans_table) == account_number_)
```

**Review by ecino — CHANGES_REQUESTED** [review]
Please move the code into the module `compassion-accounting/l10n_ch_import_camt054/models/custom_parser.py` so that we have this change in our code instead of OCA.

File: account_bank_statement_import_camt_oca/models/parser.py (hunk 2 above)
Comment (TheRaphael0000) [reply account_bank_statement_import_camt_oca/models/parser.py:210]: In my database it's not set for every record

image

**Pull request closed without merging**